services/httpd/handler.go

```diff
@@ -292,11 +292,18 @@ func (h *Handler) serveQuery(w http.ResponseWriter, r *http.Request, user *meta.
 
 	// Make sure if the client disconnects we signal the query to abort
 	closing := make(chan struct{})
+	reqDone := make(chan struct{})
```

Comment:
Also, can you only do this when a `CloseNotifier` is used? I think `CloseNotify` is always used in practicality, but I would like that section of the code moved into the `if` statement anyway with a comment about why it is needed.